src/components/ImageListContainer.js

```diff
@@ -0,0 +1,81 @@
+import React from "react";
+import {addImages} from "../actions";
+import {connect} from "react-redux";
+import ImageList from "./ImageList";
+import PropTypes from "prop-types";
+
+const IMAGE_HEIGHT = 200;
+
+class ImageListContainer extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            isFetching: false,
+        };
+    }
+
+    handleScroll = (e) => {
+        const bottom = e.target.scrollingElement.scrollHeight - IMAGE_HEIGHT - e.target.scrollingElement.scrollTop <= e.target.scrollingElement.clientHeight + 1;
+        if (bottom) {
+            this.addCat();
+        }
+    }
+
+    componentDidMount() {
+        this.addCat();
+        window.addEventListener("scroll", this.handleScroll);
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener("scroll", this.handleScroll);
+    }
+
+    addCat = () => {
+        if (!this.state.isFetching) {
+            this.setState({
+                isFetching: true
+            });
+            fetch("https://api.thecatapi.com/v1/images/search?limit=10")
```

Comment:
It is more common to use `redux-thunk` for such cases: https://redux.js.org/usage/writing-logic-thunks